### ISSUE TITLE
Project status

### ISSUE THREAD

**Issue by ghost** [squentin/gmusicbrowser#149]
What is gmusicbrowser's status? The last release was 2 years ago but I see there have been some commits made since then.

I really like it, it's the only music player on linux that has all the features I need.

**Comment by squentin** [comment]
I'm sorry, I'm having trouble finding time to get back to it.  
I'll be less busy in the coming weeks, so I'll try to finally make a new release. And then maybe finish my work on multi-song files.  
I'll close this when I actually make a new release.

**Comment by fbg13** [comment]
Thanks.

What about the migration to GTK3, will it ever happen?

**Comment by squentin** [comment]
Yes, gtk3 is a high priority, but it's also a lot of work, so I might work on some small things first. But for now, I try to focus on one thing at a time.

Also, I am using gmb everyday, and can't imagine using anything else, so at the very least I'll always keep it running.

**Comment by fbg13** [comment]
Glad to hear.

**Comment by chinarulezzz** [comment]
+1 my everyday player. Don't leave it please.

**Comment by diontje** [comment]
+1 
i love this player
submitted a small donation
thanks for your work on this and the support of PERL :)

**Comment by chinarulezzz** [comment]
Also, about donations: what about cryptocurrencies?

**Comment by squentin** [comment]
You can send bitcoins to this address : 1EEoNEiXYoeo8dZrq6syGft52tHfNp2P7r

Ask me if you want to send other cryptocurrencies, I'm not very familiar all this, I just have an account at coinbase.

**Comment by thwaller** [comment]
Wanted to add to the support for an updated release.

**Comment by RomanHargrave** [comment]
> Yes, gtk3 is a high priority, but it's also a lot of work,  so I might work on some small things first. But for now, I try to focus  on one thing at a time.

@squentin due to my avoidance of GTK3 in favor of other toolkits (I'll even take TK) I (and many others, especially those that use something like GMusicBrowser which is customizable) would probably prefer that if a toolkit migration were made that it would be to a toolkit abstraction layout (something that allows you to select your prefered toolkit) or optimally Qt since it seems to be the great defuser of the toolkit wars.

**Comment by MigueLesPaul** [comment]
I love this so much. I just wanted to be proficient developing Perl code just to work in this. This is the best linux media player ever and It runs  even faster than competition for me ! !

**Comment by fbg13** [comment]
@squentin Any progress towards a new release?

Any reason not to make a release of the current master? I've been using the master version since i opened this issue and have not encountered any problems.

I packaged gmusicbrowser for Solus and would like to use the official package since the current release is missing the "Simplify tree" feature.

**Comment by squentin** [comment]
> Any reason not to make a release of the current master?

I just want to do some small things like update translations, update search picture parsers and maybe add a patch or two that were submitted, but I have trouble finding time and for some annoying stuff like translations, motivation. I'll try to do it this month.

> @squentin due to my avoidance of GTK3 in favor of other toolkits (I'll even take TK) I (and many others, especially those that use something like GMusicBrowser which is customizable) would probably prefer that if a toolkit migration were made that it would be to a toolkit abstraction layout (something that allows you to select your prefered toolkit) or optimally Qt since it seems to be the great defuser of the toolkit wars.

As much work as migrating to gtk3/gtk4 is, it's much easier than migrating to another toolkit, so it's not going to happen. Why do you avoid gk3 ? I personally don't like what gnome has become, but gtk3 seems alright.

As for a toolkit abstraction layout, no idea what currently exist, but this kind of thing only work well for simple stuff. Some parts of gmb are very elaborate, in particular the songtree that tries very hard to look like a standard gtktreeview widget, and could not work with an abstraction layout.

**Comment by RomanHargrave** [comment]
@squentin 

> As much work as migrating to gtk3/gtk4 is, it's much easier than migrating to another toolkit, so it's not going to happen. Why do you avoid gk3 ? I personally don't like what gnome has become, but gtk3 seems alright.

GTK3 introduced a few new "idiosyncrasies". In particular, the new file picker is one of the most criminally awful file pickers I have ever seen or used. I do not like how it performs a recursive search when you begin typing, as opposed to the previous behavior of selecting a file ***immediately within the directory*** based on a name match. I created an issue on the GTK3 tracker requesting that maybe an option be added to disable this behavior, as I strongly dislike it, and received what was (if I recall correctly) an _extremely_ condescending response and closure (which is apparently not uncommon). Additionally, many of the prefabricated toolkit windows have their own decorations that are different from the environment I use, and the new design guidelines that the project has been following seem to encourage wasting space in visually unpleasant ways (trying to mimic software for macOS). Honestly, the only thing that GTK3 has going for it IMO is that it makes theme development a lot easier.

**Comment by bor** [comment]
Is not the time for v1.1.16 yet?

**Comment by dagobertstyle** [comment]
Hi, any news about an update ?

**Comment by RomanHargrave** [comment]
@squentin did respond to an issue recently

**Comment by gzsombor** [comment]
Now, the last release happened 4.5 years ago. I can understand the lack of time and motivation could be a serious issue, but maybe there are other people out there, who have these resources,and could work on the project, if they've got access. Unfortunately I'm in the camp of people who 'lack of time' and 'lack of Perl/desktop development knowledge', so I can't promise anything.

**Comment by squentin** [comment]
Good news, I've finally started to port it to gtk3 last week :)
I'm a bit forced to do it, as ubuntu (at least) has already removed gtk2, so I have to do it if I want to keep my distribution updated.
It's going well so far, I've managed to disable all calls to gtk2 with minimal changes, so I can test each part as I port it. I have a few basic layout widgets working, with their menus, and 4 tabs of the settings window. I'm a bit afraid of hitting problems with the SongList and SongTree, but we'll see then.
